### ISSUE TITLE
Could I choose ROI for stereo initialization？

### ISSUE THREAD

**Issue by CaesarTheFox** [dicengine/dice#130]
My problem is a little like #120. Because part of the picture is not useful and some random pattern may cause large error such as glare. When I do some tests about cross-correlation. I found the cropped pictures have much better result if only considering cross-correlation. The **Match** in Paraview shows the cross-correlation result. 
But the problem is that when I crop the picture, it change the coordinate of each pixels and though I get good cross correlation, the model coordinate is wrong.

Cropped picture
![fm_projective_trans_0](https://user-images.githubusercontent.com/46268648/60113172-1cad7200-9769-11e9-8a2e-f8fa59c0ad87.png)

Original picture
![fm_projective_trans_0](https://user-images.githubusercontent.com/46268648/60113254-48305c80-9769-11e9-8d90-48ed09164ebb.png)
(There is a glare causing incorrect projection and it ruins the result.)

These are _fm_projective_trans_0.png_. I guess this is to calculate the 8 parameters thing we discussed last time. And _right_projected_to_left_color.tif_ is the result of right plane projected to left plane with those 8 parameters.

So how can I draw ROI for this initilazition step?  Is this possible? Or is it possible to use exhaustive search for cross-correlation? Because I struggle with cross correlation now.

**Comment by CaesarTheFox** [comment]
In my understanding of cross-correlation prcedure. The initialization is really important in cross-correlation because if the initial guess is too far away from real value, the Newton-Raphson method will never jump there (jump_tolerance is only to avoid nonsense result instead of jump to find true value). 

Am I correct? So if I want to improve cross-correlation, it is necessary to get a better initial guess. How close should the initial guess be away from real value?

**Comment by dicengine** [comment]
You are correct about the initial guess, which is sometimes hard in stereo correlation. If you want a good cross-correlation, you need a good initial guess. The initial guess can be roughly 3-5pixels off and will often still converge. Having a better cross-correlation initialization method in DICe is something that is needed. For example if you have a non-planar surface, the cross-correlation will likely fail.

On Jun 25, 2019, at 11:48 AM, CaesarTheFox <notifications@github.com<mailto:notifications@github.com>> wrote:


In my understanding of cross-correlation prcedure. The initialization is really important in cross-correlation because if the initial guess is too far away from real value, the Newton-Raphson method will never jump there (jump_tolerance is only to avoid nonsense result instead of jump to find true value).

Am I correct? So if I want to improve cross-correlation, it is necessary to get a better initial guess. How close should the initial guess be away from real value?

**Comment by CaesarTheFox** [comment]
So could I choose ROI for stereo initialization？It influences the initialization significantly. Now I need to mask the part I dont want in Photoshop before processing. If we can have such a feature it would be more convenient I guess.
